NewForce-at-Mountwest/bangazon-api-the-chaos-whales
# Access 'Product Type' resource

# Description

I created a ProductTypesController.cs file to allow developers to access the Product Type resource through a Web API. The verbs supported by this file are GET, POST, PUT, DELETE. 
+ A user can GET a list of product types, and GET a single product type by its Id.
+ A user can POST to add a new product type.
+ A user can PUT to edit the name of an existing product type.
+ A user can DELETE an existing product type.

Before starting on ProductTypesController.cs, I made sure my previously seeded data for the ProductType table in the SQL database was still there and did not mysteriously magic itself into a black hole.

In the TestBangazonAPI project, I added appropriate code in the DatabaseFixture.cs file to make my ProductTypeTest.cs file run accordingly with the rest of the tests.
+ I added tests in ProductTypeTest.cs to make sure my GET, POST, PUT, and DELETE methods were working properly in my API.

## Related Ticket(s)
https://trello.com/b/maFChthL/the-chaos-whales
+ Ticket #5- Allow developers to access the Product Type resource

## Steps to Test
1. After making sure you have our repo cloned down into a local repository, `git fetch --all` in the `master` branch to retrieve this branch named `sm-productTypeBranch` that I have pushed up.

2. `git checkout sm-productTypeBranch` to switch to my branch and open up Visual Studio

3. If you still haven't seeded your database with this data, how did you you even make it this far into our project?! JK, nonetheless, seed your database with this data that has not changed for the past 4 tickets.

+ Follow this link to see the SQLQuery file with the data: 
    https://github.com/NewForce-at-Mountwest/bangazon-api-the-chaos-whales/blob/sm-productTypeBranch/SQLQuery.sql

***Be sure to insert Customers first, then PaymentType, then ProductType, then Product, then [Order], and finally OrderProduct (comment out only the chunk of inserts you want to add at a time)

4. Then, run the ProductTypesController.cs to check that a browser window opens with `/api/producttypes`, which should show 3 product types.

5. Run a GET single product type by using `/api/producttypes/2` to see a product type with a Name of "Porcelain Mug"

6. Next, hop on over to the TestBangazonAPI project and open up the Test Explorer. Select 'ProductTypeTest' in the list of folders available to test, and select 'Run Test' or the green play button.

7. If all the tests (5 tests) pass successfully, marked by a green circle with a checkmark, but don't you celebrate just yet.

8. Now, you need to "Run All Test(s)" to see if this branch's code works with the rest of the code because that would matter A LOT. Once all the tests run successfully and are marked by reinforcing green circles and checkmarks, feel free to celebrate this branch's functionality.


# Checklist:
- [ ] I have opened up `sm-productTypeBranch` in my local repository.
- [ ] I have seeded my SQL database with the data provided above, if applicable.
- [ ] I have opened up the localhost browser window to test the get and get by Id method (i.e. "api/producttypes", "api/producttypes/1", "api/producttypes/2", "api/producttypes/3")
- [ ] I have hopped over to TestBangazonAPI and ran all 5 relevant 'ProductType' tests and made sure they passed.
- [ ] I have selected "Run All Test(s)" in the Test Explorer and made sure they all passed.
- [ ] I have celebrated this code's functionality with a "Hip Hip Hooray" and if available, an enthusiastic throw of confetti!
